drivers/storage/portworx/deployment.go

```diff
@@ -0,0 +1,636 @@
+	}
+
+	if t.isPxDev {
+		args = append(args, "--dev")
+	}
+
+	if t.isPKS {
+		args = append(args, "--keep-px-up")
+	}
+
+	if t.cluster.Annotations[annotationLogFile] != "" {
+		args = append(args, "--log", t.cluster.Annotations[annotationLogFile])
+	}
+
+	if t.cluster.Annotations[annotationMiscArgs] != "" {
+		parts, err := shlex.Split(t.cluster.Annotations[annotationMiscArgs])
+		if err == nil {
+			args = append(args, parts...)
+		} else {
+			logrus.Warnf("error parsing misc args: %v", err)
+		}
+	}
+
+	return args
+}
+
+func (t *template) getEnvList() []v1.EnvVar {
+	envList := []v1.EnvVar{
+		{
+			Name:  "AUTO_NODE_RECOVERY_TIMEOUT_IN_SECS",
+			Value: "1500",
+		},
+		{
+			Name:  "PX_TEMPLATE_VERSION",
+			Value: templateVersion,
+		},
+	}
+
+	if t.isPKS {
+		envList = append(envList, v1.EnvVar{
+			Name:  "PRE-EXEC",
+			Value: "if [ ! -x /bin/systemctl ]; then apt-get update; apt-get install -y systemd; fi",
+		})
+	}
+
+	if CSI.isEnabled(t.cluster.Spec.FeatureGates) {
+		envList = append(envList, v1.EnvVar{
+			Name:  "CSI_ENDPOINT",
+			Value: "unix://" + csiBasePath + "/csi.sock",
+		})
+	}
+
+	if t.cluster.Spec.ImagePullSecret != nil && *t.cluster.Spec.ImagePullSecret != "" {
+		envList = append(envList, v1.EnvVar{
+			Name: "REGISTRY_CONFIG",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					// TODO: for older versions of k8s it could be .dockercfg
+					Key: ".dockerconfigjson",
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: *t.cluster.Spec.ImagePullSecret,
+					},
+				},
+			},
+		})
+	}
+
+	for _, env := range t.cluster.Spec.Env {
+		envCopy := env.DeepCopy()
+		envList = append(envList, *envCopy)
+	}
+
+	return envList
+}
+
+func (t *template) getVolumeMounts() []v1.VolumeMount {
+	volumeMounts := []v1.VolumeMount{
+		{
+			Name:      "diagsdump",
+			MountPath: "/var/cores",
+		},
+		{
+			Name:      "dockersock",
+			MountPath: "/var/run/docker.sock",
+		},
+		{
+			Name:      "containerdsock",
+			MountPath: "/run/containerd",
+		},
+		{
+			Name:      "etcpwx",
+			MountPath: "/etc/pwx",
+		},
+	}
+
+	if !t.isDocker {
+		volumeMounts = append(volumeMounts,
+			v1.VolumeMount{
+				Name:      "optpwx",
+				MountPath: "/opt/pwx",
+			},
+			v1.VolumeMount{
+				Name:      "procmount",
+				MountPath: "/host_proc",
+			},
+			v1.VolumeMount{
+				Name:      "sysdmount",
+				MountPath: "/etc/systemd/system",
+			},
+			v1.VolumeMount{
+				Name:      "journalmount1",
+				MountPath: "/var/run/log",
+				ReadOnly:  true,
+			},
+			v1.VolumeMount{
+				Name:      "journalmount2",
+				MountPath: "/var/log",
+				ReadOnly:  true,
+			},
+			v1.VolumeMount{
+				Name:      "dbusmount",
+				MountPath: "/var/run/dbus",
+			},
+		)
+
+		if t.isPKS {
+			volumeMounts = append(volumeMounts,
+				v1.VolumeMount{
+					Name:      "pxlogs",
+					MountPath: "/var/lib/osd/log",
+				},
+			)
+		}
+	} else {
+		// TODO: older versions of k8s don't support mountPropagation
```

Comment:
Not sure. Just left a comment in case we decide to support pre 1.10 of k8s.
I guess openshift 3.8 was using k8s 1.9 which does not have mount propagation.